tests/testthat/test-interface-accessors.R

```diff
@@ -44,3 +44,16 @@ test_that("s2_x and s2_y works", {
   expect_identical(s2_y("POINT EMPTY"), NA_real_)
 })
 
+test_that("s2_distance works", {
+  expect_equal(
+    s2_distance("POINT (0 0)", "POINT (0 90)", radius = 180 / pi),
+    90
+  )
+
+  expect_identical(s2_distance("POINT (0 0)", NA_character_), NA_real_)
+  expect_identical(s2_distance(NA_character_, "POINT (0 0)"), NA_real_)
+  # apparently Infinite is the distance returned for an empty
+  expect_identical(s2_distance("POINT (0 0)", "POINT EMPTY"), Inf)
+  expect_identical(s2_distance("POINT EMPTY", "POINT (0 0)"), Inf)
```

Comment:
I recall we had this discussion on twitter, on your wk package. I don't have strong feelings here, but still feel you're overdoing it: 
1. you introduce another geometry, whose (WKT?) representation is `NA_character_`, which results in NA values in distance computations - to which geometry in e.g. SFA or S2 frameworks does this correspond?
2. for empty geometries (the empty set), you return `Inf` in distance computations. Why not NA? To me, `Inf` suggests a point exists but is very far away. But a `POINT EMPTY` does not exist.

@rubak you are a mathematician, and aware of what spatstat does: what is your opinion on this?

Reply:
I don't think I have a better take on this than you @edzer. I also feel that `Inf` may be misleading. What is the typical usage where an empty geometry appears? Something like looping through all pairs of polygons in a list and finding their intersection? Afterwards we may query the distance from some given point to each intersection. What would we expect to happen if the intersection is empty? What is convenient when programming?

One problem with using `Inf` is that it would be natural to use `Inf` distance in extension packages where it may indicate there is no way from this point to that point due to boundaries or whatever. Then `Inf` would mean either that one (or both) of the points doesn't exist at all or that they exist but there is no way from one point to the other.

Reply:
Fair enough! I think our comments were just meant helping find a good direction: `NA_character_` cannot be a geometry, empty geometries lead to `NA` distances.

Reply:
We are all plumbes! What do you mean by "partial empty input", like at least one of the two arguments is an empty geometry?